### PR TITLE
Add random quotes to the collection selection window from a tsv file.

### PR DESCRIPTION
**Pull request by bjsi**
This is a continuation of https://github.com/supermemo/SuperMemoAssistant/pull/68. Apologies for the inconvenience, I had some issues with my git workflow that should now be resolved.

I added your suggestions and tested with a well-formatted, misformatted and missing quotes.tsv file.

One thing that has changed is the box outline from the collection selection box.
image

### CODE REVIEW

File: src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml [hunk 1]
```diff
@@ -34,6 +34,7 @@
   <Grid>
     <Grid.RowDefinitions>
       <RowDefinition Height="*" />
+      <RowDefinition Height="130"/>
```
Comment (alexis-) [inline src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml]: Why not `Height="auto"` ?

File: src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml.cs [hunk 2]
```diff
@@ -244,6 +250,61 @@ private void BtnUpdates_Click(object sender, RoutedEventArgs e)
...
+                                   RoutedEventArgs e)
+    {
+      // Tab separated file with a heading
+      // Quote, Author, Url, Title
+      var quoteFile = SMAFileSystem.AppRootDir.CombineFile("quotes.tsv");
+      if (quoteFile.Exists())
+      {
+        try
+        {
+          var lines = File.ReadAllLines(quoteFile.FullPath);
```
Comment (alexis-) [inline src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml.cs]: That's not going to work with the latest version. You have to use `SMAFileSystem.GetAppExeFilePath` to combine your file's name with the executed app's directory.

`SMAFileSystem.AppRootDir` will return e.g. `C:\Users\Jamesb\AppData\Local\SuperMemoAssistant`
`SMAFileSystem.GetAppExeFilePath` will combine your file with e.g. `C:\Users\Jamesb\AppData\Local\SuperMemoAssistant\app-2.0.4.84`

File: src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml.cs [hunk 3]
```diff
@@ -244,6 +250,61 @@ private void BtnUpdates_Click(object sender, RoutedEventArgs e)
...
+          var randInt = new Random();
+          var randomLineNumber = randInt.Next(1, lines.Length - 1);
+          var quoteLine = lines[randomLineNumber];
+          var splitQuoteLine = quoteLine.Split('\t');
+
+          // Check that the chosen line has 4 fields
+          if (splitQuoteLine.Length == 4)
+          {
+            // If the quote doesn't end with sentence ending punctuation, add a full stop.
+            if (!SentenceEndingPunctuation.Any(p => splitQuoteLine[0].EndsWith(p)))
```
Comment (alexis-) [inline src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml.cs]: Fancy

File: src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml (hunk 1 above)
Comment (bjsi) [reply src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml]: Ah yeah that's a better solution!

File: src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml.cs (hunk 2 above)
Comment (bjsi) [reply src/AppHosts/SuperMemoAssistant/UI/CollectionSelectionWindow.xaml.cs]: Ok noted.

**Pull request merged**